armi/reactor/blueprints/componentBlueprint.py

```diff
@@ -225,10 +225,10 @@ def _constructMaterial(self, blueprint, matMods):
             try:
                 # update material with updated input params from blueprints file.
                 mat.applyInputParams(**matMods)
-            except TypeError:
+            except TypeError as te:
                 # This component does not accept material modification inputs of the names passed in
                 # Keep going since the modification could work for another component
-                pass
+                runLog.warning("Error constructing {}: {}".format(mat, te))
```

Comment:
So this is a good start, but there are a couple of peculiarities with how we handle material modifications that make this function a bit differently than you might expect. For a given Assembly blueprint, I think all of the material modifications for that assembly get passed to all of the materials that make up that assembly. So if you have a material modification for, say Uranium enrichment, it would get passed to the other materials that dont know anything about U235 weight fraction, and they would all produce warnings. This might become more clear if you add some other components to your block design in the test.

We were talking about this some the other day, and I think ive got an interesting approach for this. I'll comment it in #402

Reply:
Thank you, I did actually debug this with more material modifications in the blueprint but hadn't investigated the myriad of warnings enough to appreciate this peculiarity you mentioned... I did notice that the way a material's "applyInputParams" method (such as uZr.py) directly calls `material.FuelMaterial.applyInputParams(self, *args, **kwargs)` instead of calling up the class hierarchy.  I think refactoring this, and doing input validation on these material modification parameters,, might make for a clearer design.  I'm glad that you propose a more robust solution and will continue the discussion on #402.